qiskit_dynamics/solvers/dyson_magnus_solvers.py

```diff
@@ -0,0 +1,995 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+# pylint: disable=invalid-name
+
+"""
+Model class for perturbative expansion of solution of an LMDE.
+"""
+
+from abc import ABC, abstractmethod
+from typing import Optional, Tuple, Callable, List, Union
+
+import numpy as np
+from numpy.polynomial.chebyshev import chebpts1, chebvander, chebval
+from scipy.linalg import expm
+from scipy.integrate._ivp.ivp import OdeResult
+
+from multiset import Multiset
+
+from qiskit import QiskitError
+from qiskit.quantum_info import Operator
+
+from qiskit_dynamics import Signal, RotatingFrame
+from qiskit_dynamics.signals import SignalList
+from qiskit_dynamics.perturbation import solve_lmde_perturbation, ArrayPolynomial
+from qiskit_dynamics.array import Array
+from qiskit_dynamics.type_utils import to_array
+from qiskit_dynamics.dispatch.dispatch import Dispatch
+
+from .solver_utils import setup_args_lists
+
+try:
+    import jax.numpy as jnp
+    from jax import vmap
+    from jax.scipy.linalg import expm as jexpm
+    from jax.lax import scan, associative_scan
+except ImportError:
+    pass
+
+
+class PerturbativeSolver(ABC):
```

Comment:
Can you break this up into 4 separate files for the 4 main classes: PertubativeSolver, MagnusSolver and DysonSolver, ExpansionModel? Might be good to put them in a subfolder for organization too.